tests/test_core.py

```diff
@@ -5757,7 +5749,6 @@ def test_autodebug(self):
 
     # Compare to each other, and to expected output
     self.do_ll_run(path_from_root('tests', filename+'.o.ll.ll'), '''AD:-1,1''')
-    assert open('stdout').read().startswith('AD:-1'), 'We must note when we enter functions'
```

Comment:
I should probably add comments to all of these changes, because they're generally pretty weird.

For this one, the assert is subsumed by the previous line that checks that `'AD:-1,1'` shows up in the output somewhere. The line was added before the `do_ll_run` call was checking the output, and hasn't changed since.
The reason this was failing only with v8 here was that it was printing the "trying binaryen method" header first. This is v8-only because v8 doesn't have printErr, and so gets routed to print. This means it passes the "is the string somewhere in the output" test, but fails the "is the string right at the beginning of the output" test.
[I made a patch in v8 to add printErr](https://codereview.chromium.org/2458963003/) because that turns out to be incredibly convenient for debugging compiler.js, jsifier.js, etc. Not sure if/when that'll go in, or if we have some more fundamental reason for not having printErr in v8 in the first place.
Either way, the `startswith` test is over-specific - nothing else in test_core seems to care about where the expected output is found - and redundant, so we should remove it as well.